Source/WarpX.cpp

```diff
@@ -1033,16 +1033,16 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     }
     bool const pml_flag_false=false;
     spectral_solver_fp[lev].reset( new SpectralSolver( realspace_ba, dm,
-        nox_fft, noy_fft, noz_fft, do_nodal, m_v_galilean, dx_vect, dt[lev],
+        nox_fft, noy_fft, noz_fft, do_cell_centered, m_v_galilean, dx_vect, dt[lev],
         pml_flag_false, fft_periodic_single_box, update_with_rho, fft_do_time_averaging ) );
 #   endif
 #endif
     m_fdtd_solver_fp[lev].reset(
-        new FiniteDifferenceSolver(maxwell_solver_id, dx, do_nodal) );
+        new FiniteDifferenceSolver(maxwell_solver_id, dx, do_cell_centered) );
     //
     // The Aux patch (i.e., the full solution)
     //
-    if (aux_is_nodal and !do_nodal)
+    if (aux_is_nodal and !do_cell_centered)
     {
         // Create aux multifabs on Nodal Box Array
         BoxArray const nba = amrex::convert(ba,IntVect::TheNodeVector());
```

Comment:
Note: this might need to be changed too (IndexType of "aux" MultiFabs for momentum-conserving field gathering)
```suggestion
        BoxArray const nba = amrex::convert(ba,IntVect::TheCellVector());
```